reference/reference_test.go

```diff
@@ -96,6 +96,23 @@ func TestReferenceParse(t *testing.T) {
 			err:   digest.ErrDigestUnsupported,
 		},
 		{
+			input: "Uppercase:tag",
+			err:   ErrNameContainsUppercase,
+		},
+		{
+			input: "Uppercase/lowercase:tag",
+			err:   ErrNameContainsUppercase,
```

Comment:
@aaronlehmann what do you prefer; shall I remove the failing test, or comment it and add a TODO?